### DISCUSSION TITLE
Can i pass custom data while submitting forms?

### DISCUSSION THREAD

**Discussion by MadhusudhanReddyWeWork** [Budibase/budibase#13675] [Help] [answered]
I wanted to send some metadata from the application in which i am using budibase forms so that the metadata will also be saved in the DB. is it possible to do so? i have explored actions but that didn't help

**Comment by mikesealey** [comment, marked as answer]
Hi @MadhusudhanReddyWeWork, I've had a look at this for you.

Budibase supports using `JSON` as a column type in the tables, which may be how you want to store your metadata, but you could also create a new column type for each metadata value you wish to store. 

When using a `Form block` component, select your table `Data `source. This will populate all columns in the table as fields in the form. Turn off the metadata column so that it does not appear to the user. 
![image](https://github.com/Budibase/budibase/assets/39269380/4e929214-ff54-4177-be87-3b71464761a6)

Then, in the actions triggered by the Save button, head to the Save Row action and click "Add column", select your metadata column(s) and set your values there. You may wish to make use of `State` to help you pass values around from different areas of your application - you can read more about `State` [here in the documentation.](https://docs.budibase.com/docs/app-state).

![image](https://github.com/Budibase/budibase/assets/39269380/b2345e12-cb87-4c90-b285-6cbcf406a8bb)

Note: Budibase currently provides five auto-columns in the Data section for you which may also be useful for your application.
![image](https://github.com/Budibase/budibase/assets/39269380/442e9dfb-055d-4773-9c67-049e38c2e1f0)

Hopefully that should take care of your needs. If you haven't already, you could also join our [community discord server](https://discord.com/invite/ZepTmGbtfF), where you can discuss ideas and solutions in _Community Help_, or share your work for bragging rights in _Showcase_.

**Reply by MadhusudhanReddyWeWork** [reply]
thanks @mikesealey  for the suggestion, but i have already checked this out, if i want the metadata to be populated from the external application(application in which budibase is being used) and not inside the budibase application, is it possible to populate state with the data from the external/parent application?
